dias/analyzers/flag_rfi_analyzer.py

```diff
@@ -0,0 +1,592 @@
+                                               unit="seconds")
+
+        # Add data metrics
+        self.fraction_masked_missing = self.add_data_metric(
+                                                "fraction_masked_missing",
+                                                "Fraction of data that is " +
+                                                "missing (e.g., dropped " +
+                                                "packets or down GPU nodes).",
+                                                labelnames=['stack'])
+
+        self.fraction_masked_before = self.add_data_metric(
+                                                "fraction_masked_before",
+                                                "Fraction of data considered" +
+                                                "considered bad before " +
+                                                "applying MAD threshold.  " +
+                                                "Includes missing data and " +
+                                                "static frequency mask.",
+                                                labelnames=['stack'])
+
+        self.fraction_masked_after = self.add_data_metric(
+                                                "fraction_masked_after",
+                                                "Fraction of data considered" +
+                                                "considered bad after " +
+                                                "applying MAD threshold.  " +
+                                                "Includes missing data, " +
+                                                "static frequency mask and " +
+                                                "MAD threshold mask.",
+                                                labelnames=['stack'])
+
+    def run(self):
+        """Run the task.
+
+        Load stacked autocorrelations from the last period,
+        generate rfi mask, write to disk, and update the
+        data index database.
+        """
+        self.run_start_time = time.time()
+
+        # Refresh the database
+        self.refresh_data_index()
+
+        # Determine the range of time to process
+        end_time = datetime.datetime.utcnow() - self.offset
+
+        cursor = self.data_index.cursor()
+        query = 'SELECT stop FROM files ORDER BY stop DESC LIMIT 1'
+        results = list(cursor.execute(query))
+        start_time = results[0][0] if results else end_time - self.period
+
+        self.logger.info('Analyzing data between {} and {}.'.format(
+                         datetime2str(start_time),  datetime2str(end_time)))
+
+        # Use Finder to get the files to analyze
+        try:
+            finder = self.Finder()
+            finder.accept_all_global_flags()
+            finder.only_corr()
+            finder.filter_acqs(data_index.ArchiveInst.name == self.instrument)
+            finder.set_time_range(start_time, end_time)
+
+        except Exception as exc:
```

Comment:
If you catch the exact exception type you expect, you will get a full report if something ever gets raised that you didn't expect.